### ISSUE TITLE
Draining & rebooting nodes results in DNS failures

### ISSUE THREAD

**Issue by Vandersteen** [coredns/coredns#3010]
After draining -> rebooting -> uncordoning our nodes on an aks cluster, we started to receive a whole lot of DNS failures (for a few minutes), I am trying to figure out if there is a way to prevent this from happening when doing maintenance on our cluster. As this did affect some of our applications

See the logs below for some example failures we received

Steps taken:

```
$ kubectl drain <node> --ignore-daemonsets --delete-local-data
# Note: i had to add --delete-local-data otherwise the drain would not
# take place as coredns was using local data

# reboot node
# wait for ready

$ kubectl uncordon <node>
```

CoreDns is installed like AKS configured it for us (with an autoscaler etc)

When reporting bugs and requesting assistance, please include:
* K8s version
v1.13.5

* the version of CoreDNS you are using
aksrepos.azurecr.io/mirror/coredns:1.2.6

* your Corefile

```
.:53 {
    errors
    health
    kubernetes cluster.local in-addr.arpa ip6.arpa {
      pods insecure
      upstream
      fallthrough in-addr.arpa ip6.arpa
    }
    prometheus :9153
    proxy . /etc/resolv.conf
    cache 30
    loop
    reload
    loadbalance
    import custom/*.override
}
import custom/*.server
```

* logs, if applicable
```
 [ERROR] plugin/errors: 2 _mongodb._tcp.xxxx.azure.mongodb.net. SRV: unreachable backend: no upstream host
 [ERROR] plugin/errors: 2 _mongodb._tcp.xxxx.azure.mongodb.net. SRV: unreachable backend: no upstream host
 [ERROR] plugin/errors: 2 _mongodb._tcp.xxxx.azure.mongodb.net. SRV: unreachable backend: no upstream host
...
[ERROR] plugin/errors: 2 xxxx-headless.default.svc.cluster.local.b53kgyupvfvexi5zvwwhdlzqwg.ax.internal.cloudapp.net. A: unreachable backend: no upstream host
 [ERROR] plugin/errors: 2 xxxx-headless.default.svc.cluster.local.b53kgyupvfvexi5zvwwhdlzqwg.ax.internal.cloudapp.net. AAAA: unreachable backend: no upstream host
 [ERROR] plugin/errors: 2 xxxx-headless.default.svc.cluster.local.b53kgyupvfvexi5zvwwhdlzqwg.ax.internal.cloudapp.net. AAAA: unreachable backend: no upstream host
```

Is there a way to prevent these issues ?

**Comment by chrisohaver** [comment]
The error is coming from the *proxy* plugin, which has been deprecated, and later replaced by the *forward* plugin since CoreDNS 1.2.6 was released. The error occurs when the plugin finds no upstream nameservers that are "up." This could be an artifact of how *proxy* monitors health of upstream nameservers.  Try using the *forward* plugin instead of *proxy* to see if that helps.

**Comment by Vandersteen** [comment]
Is it standard since 1.2.6 ? As we are running 1.2.6 right now.

If not, would you be so kind to point me to some documentation on how one would use the forward plugin instead of the proxy plugin on kubernetes.
Have been looking around but could not find pointers.

Or would I have to upgrade coredns to version x all together ? If so, which version would you recommend ?

**Comment by chrisohaver** [comment]
*forward* has been included in CoreDNS since 1.0.6.
We decided to use it in the default Kubernetes configuration starting k8s 1.14 (IIRC).
*proxy* was removed from coredns in 1.5.0.

To use *forward*, replace "proxy" with "forward" in the config ... i.e. replace the "proxy" line with...

```
    forward . /etc/resolv.conf
```

**Comment by Vandersteen** [comment]
Thank you for your help & time
